lib/utils/naive_navigator.cpp

```diff
@@ -1,19 +1,42 @@
 #include "naive_navigator.hpp"
 
+#include <numeric>
+
 namespace hyped::utils {
 
 NaiveNavigator::NaiveNavigator() : current_trajectory_{}
 {
 }
 
-navigation::Trajectory NaiveNavigator::currentTrajectory()
+std::optional<core::Trajectory> NaiveNavigator::currentTrajectory()
 {
   return current_trajectory_;
 }
 
-void NaiveNavigator::update(const navigation::Trajectory &current_trajectory)
+void NaiveNavigator::keyenceUpdate(const core::KeyenceData &keyence_data)
+{
+  // Do nothing. Keyence has no direct influence on trajectory
+}
+
+void NaiveNavigator::encoderUpdate(const core::EncoderData &encoder_data)
+{
+  core::Float sum                  = std::accumulate(encoder_data.begin(), encoder_data.end(), 0.0);
+  core::Float encoder_average      = static_cast<core::Float>(sum / core::kNumEncoders);
+  current_trajectory_.displacement = encoder_average;
+}
+
+void NaiveNavigator::imuUpdate(const core::RawImuData &imu_data)
 {
-  current_trajectory_ = current_trajectory;
+  core::Float sum = 0.0;
+  for (size_t i = 0; i < core::kNumImus; ++i) {
+    for (size_t j = 0; j < core::kNumAxis; ++j) {
+      sum += imu_data.at(i).at(j);
+    }
+  }
+  core::Float imu_average          = static_cast<core::Float>(sum / core::kNumImus);
```

Comment:
White space?

Reply:
Auto formatting is happy so we're all good